pennylane_lightning/src/simulator/StateVectorManaged.hpp

```diff
@@ -36,47 +36,57 @@ class StateVectorManaged
     std::vector<ComplexPrecisionT> data_;
 
   public:
-    StateVectorManaged() : StateVectorBase<PrecisionT, StateVectorManaged>() {}
-
+    /**
+     * @brief @rst
+     * Create a new statevector whose data is managed by C++. The state
+     * is initialized as :math:`|0\rangle^{\otimes N}` where :math:`N` is
+     * the number of qubits.
+     * @endrst
+     *
+     * @param num_qubits Number of qubits
+     */
     explicit StateVectorManaged(size_t num_qubits)
         : BaseType(num_qubits),
           data_(static_cast<size_t>(Util::exp2(num_qubits)),
                 ComplexPrecisionT{0, 0}) {
         data_[0] = {1, 0};
     }
 
+    /**
+     * @brief Construct a statevector from a data vector
+     */
     template <class OtherDerived>
     explicit StateVectorManaged(
         const StateVectorBase<PrecisionT, OtherDerived> &other)
         : BaseType(other.getNumQubits()), data_{other.getData(),
                                                 other.getData() +
                                                     other.getLength()} {}
 
+    /**
+     * @brief Construct a statevector from a data vector
+     */
     explicit StateVectorManaged(
         const std::vector<ComplexPrecisionT> &other_data)
         : BaseType(Util::log2(other_data.size())), data_{other_data} {
         PL_ABORT_IF_NOT(Util::isPerfectPowerOf2(other_data.size()),
                         "The size of provided data must be a power of 2.");
     }
 
+    /**
+     * @brief Construct a statevector from a data pointer
+     */
     StateVectorManaged(const ComplexPrecisionT *other_data, size_t other_size)
         : BaseType(Util::log2(other_size)), data_{other_data,
                                                   other_data + other_size} {
         PL_ABORT_IF_NOT(Util::isPerfectPowerOf2(other_size),
                         "The size of provided data must be a power of 2.");
     }
 
-    StateVectorManaged(const StateVectorManaged<PrecisionT> &other) = default;
-    StateVectorManaged(StateVectorManaged<PrecisionT> &&other) noexcept =
-        default;
-
-    ~StateVectorManaged() = default;
```

Comment:
~~What was the issue with these?~~ Got my [answer](https://github.com/PennyLaneAI/pennylane-lightning/pull/274/files#r847448588)! Nice!

Reply:
:+1: